Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/ProjectionIntegrityViolationDetector.php

```diff
@@ -182,20 +182,26 @@ public function subtreeTagsAreInherited(): Result
         // NOTE:
         // This part determines if a parent hierarchy relation contains subtree tags that are not existing in the child relation.
         // This could probably be solved with JSON_ARRAY_INTERSECT(JSON_KEYS(ph.subtreetags), JSON_KEYS(h.subtreetags) but unfortunately that's only available with MariaDB 11.2+ according to https://mariadb.com/kb/en/json_array_intersect/
-        $hierarchyRelationsWithMissingSubtreeTags = $this->client->getConnection()->executeQuery(
-            'SELECT
-              ph.name
-            FROM
-              ' . $this->tableNamePrefix . '_hierarchyrelation h
-              INNER JOIN ' . $this->tableNamePrefix . '_hierarchyrelation ph
-                ON ph.childnodeanchor = h.parentnodeanchor
-                AND ph.contentstreamid = h.contentstreamid
-                AND ph.dimensionspacepointhash = h.dimensionspacepointhash
-            WHERE
-              EXISTS (
-                SELECT t.tag FROM JSON_TABLE(JSON_KEYS(ph.subtreetags), \'$[*]\' COLUMNS(tag VARCHAR(30) PATH \'$\')) t WHERE NOT JSON_EXISTS(h.subtreetags, CONCAT(\'$.\', t.tag))
-              )'
-        )->fetchAllAssociative();
+        $query = 'SELECT
+          ph.name
+        FROM
+          ' . $this->tableNamePrefix . '_hierarchyrelation h
+          INNER JOIN ' . $this->tableNamePrefix . '_hierarchyrelation ph
+            ON ph.childnodeanchor = h.parentnodeanchor
+            AND ph.contentstreamid = h.contentstreamid
+            AND ph.dimensionspacepointhash = h.dimensionspacepointhash
+        WHERE
+          EXISTS (
+            SELECT t.tag FROM JSON_TABLE(JSON_KEYS(ph.subtreetags), \'$[*]\' COLUMNS(tag VARCHAR(30) PATH \'$\')) t WHERE NOT JSON_EXISTS(h.subtreetags, CONCAT(\'$.\', t.tag))
+          )';
+
+        try {
+            $hierarchyRelationsWithMissingSubtreeTags = $this->client->getConnection()->executeQuery($query)->fetchAllAssociative();
+        } catch (\Doctrine\DBAL\Exception\SyntaxErrorException $syntaxErrorException) {
+            // Hierarchy relation subtree tags could not be validated as the database doesn't support "JSON_TABLE" feature.
+            // TODO drop this check once the db requirements are clear: https://github.com/neos/neos-development-collection/issues/4337
+            return $result;
+        }
```

Comment:
This will forever render this test useless if we happen to have a typo in the query above.
We should instead:
a) rewrite the query such that it works on older versions (maybe with a slight performance penalty)
b) ...or require MariaDB 10.6+ (which is required for [JSON_TABLE](https://mariadb.com/kb/en/json_table/) to work) – this version was released [3 years ago](https://mariadb.com/kb/en/mariadb-1060-release-notes/)